File: packages/android/src/main/java/com/kozmos/components/Navbar/Navbar.figma.kt
package com.kozmos.components.navbar

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.components.button.KozmosButton

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6765")
class KozmosNavbarConnect {
    @FigmaProperty(FigmaType.Enum, "Content")
    val content: String = Figma.mapping(
        "Basic" to "basic",
        "Actions" to "actions",
        "Contextual" to "contextual"
    )

    @Composable
    fun ComponentExample() {
        KozmosNavbar(
            logo = {
                Text("Logo")
            },
            context = {
                if (content == "contextual") {
                    Text("Context")
                }
            },
            navigation = {
                if (content == "contextual") {
                    Text("Navigation")
                }
            },
            primaryAction = {
                if (content == "contextual") {
                    KozmosButton(onClick = {}) {
                        Text("Primary action")
                    }
                }
            },
            actions = {
                if (content == "actions" || content == "contextual") {
                    Text("Actions")
                }
            },
            utilities = {
                if (content == "contextual") {
                    Text("Utility")
                }
            },
            account = {
                if (content == "contextual") {
                    Text("Account")
                }
            }
        )
    }
}

File: packages/ios/Sources/Components/Navbar/Navbar.figma.swift
import SwiftUI
import Figma

struct KozmosNavbarBasicConnect: FigmaConnect {
    let component = KozmosNavbar.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6765"
    var variant = ["Content": "Basic"]

    var body: some View {
        KozmosNavbar(
            logo: {
                Text("Logo")
            },
            context: {
                EmptyView()
            },
            navigation: {
                EmptyView()
            },
            primaryAction: {
                EmptyView()
            },
            actions: {
                EmptyView()
            },
            utilities: {
                EmptyView()
            },
            account: {
                EmptyView()
            }
        )
    }
}

struct KozmosNavbarActionsConnect: FigmaConnect {
    let component = KozmosNavbar.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6765"
    var variant = ["Content": "Actions"]

    var body: some View {
        KozmosNavbar(
            logo: {
                Text("Logo")
            },
            context: {
                EmptyView()
            },
            navigation: {
                EmptyView()
            },
            primaryAction: {
                EmptyView()
            },
            actions: {
                Text("Actions")
            },
            utilities: {
                EmptyView()
            },
            account: {
                EmptyView()
            }
        )
    }
}

struct KozmosNavbarContextualConnect: FigmaConnect {
    let component = KozmosNavbar.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6765"
    var variant = ["Content": "Contextual"]

    var body: some View {
        KozmosNavbar(
            logo: {
                Text("Logo")
            },
            context: {
                VStack(alignment: .leading) {
                    Text("Context")
                }
            },
            navigation: {
                Text("Navigation")
            },
            primaryAction: {
                KozmosButton("Primary action", action: {})
            },
            actions: {
                Text("Actions")
            },
            utilities: {
                Text("Utility")
            },
            account: {
                Text("Account")
            }
        )
    }
}
